src/expression.rs

```diff
@@ -25,6 +25,9 @@ impl Expression {
 
     // Get `Expression` by parsing a string
     pub fn from_str(s: &str) -> Result<Self, Error> {
+        if s.len() == 0 {
+            return Ok(Expression::new());
+        }
```

Comment:
Does the expression parser not handle end of input? This special case separation between parser and from_str behavior seems strange.